src/index.tsx

```diff
@@ -42,20 +34,26 @@ function concatTranscripts(...parts: string[]) {
 }
 
 export function useSpeechRecognition(options: SpeechRecognitionOptions = defaultOptions): SpeechRecognitionUtils {
-  const BrowserSpeechRecognition =
-    typeof window !== "undefined" &&
-    (window.SpeechRecognition ||
-      window.webkitSpeechRecognition ||
-      window.mozSpeechRecognition ||
-      window.msSpeechRecognition ||
-      window.oSpeechRecognition);
-
   const [
-    { listening, status, pauseAfterDisconnect, interimTranscript, finalTranscript, transcript },
+    { error, status, pauseAfterDisconnect, interimTranscript, finalTranscript, transcript },
     dispatch,
   ] = useReducer(speechRecognitionReducer, initialState);
 
-  const recognition = BrowserSpeechRecognition ? new BrowserSpeechRecognition() : undefined;
+  const recognition = useMemo(() => {
+    const BrowserSpeechRecognition =
+      typeof window !== "undefined" &&
+      (window.SpeechRecognition ||
+        window.webkitSpeechRecognition ||
+        window.mozSpeechRecognition ||
+        window.msSpeechRecognition ||
+        window.oSpeechRecognition);
+
+    if (BrowserSpeechRecognition) {
+      return new BrowserSpeechRecognition();
+    } else {
+      throw new Error(ERROR_NO_RECOGNITION_SUPPORT);
+    }
+  }, []);
 
   const disconnect = useCallback(
     (disconnectType: SpeechRecognitionDisconnectType) => {
```

Comment:
this function has a couple multiple dispatch cases which cna probably simplified to 

```
dispatch(reset());
```

and 
```
dispatch(disconnect(SpeechRecognitionStatus.STOPPED)
```

moving the state transition logic to the reducer